Added observation sessions api endpoint.

marc opened the pull request.

marc committed:
Added observation sessions api endpoint.

rposborne reviewed (changes requested):
Lets do a has_many :observations in this case, from our conversation it looks like we have a race condition if we send incomplete jsonb arrays.

marc committed:
Moved Observations into a separate model.

marc committed:
Mini code cleanup.

rposborne committed:
Refactor get_session_uuid to session_uuid

rposborne committed:
Refactor ApiObservations Controller to perform upsert like behavior.

This handles fire and forget from the client so that if a observation is sent twice, data is cleanly persisted.

rposborne committed:
Fixed foreign key to be uuid for observation_session

rposborne committed:
Set travis to use postgres 9.6
